scripts/extractStrings.js

```diff
@@ -84,44 +122,95 @@ function openFile(file) {
     const data = fs.readFileSync(file, Encoding);
     return data;
   } catch (err) {
-    console.error("Cannot open file: ", file, "ERROR: ", err);
+    console.error("ERROR: Cannot open file: ", file, "ERROR: ", err);
   }
 }
 
-function getLocalizeableStrings(obj, key, outputMap, filename) {
-  for (var i in obj) {
-    if (!obj.hasOwnProperty(i)) {
-      continue;
-    }
-    if (typeof obj[i] === 'object') {
-      getLocalizeableStrings(obj[i], key.concat(i, '.'), outputMap, filename);
-    } else if (filename.localeCompare(SettingsFile) === 0) {
-      // Settings file has different loc keys than template files
-      if (SettingsFileLocKeys.includes(i)) {
-        const jsonKey = key.concat(i);
-        outputMap[jsonKey] = obj[i];
+function getLocalizeableStrings(obj, key, outputMap) {
+  for (var field in obj) {
+    var objectEntry = obj[field];
+    var jsonKey;
+    if (typeof objectEntry === 'object') {
+      // If the last field is a number, it is part of an array.
+      // See if there's another identifier such that if a template order is edited, the string does not need to be re-localized
+      if (!isNaN(parseInt(field))) {
+        jsonKey = getKeyForArrayObject(key, objectEntry, field);
+      } else {
+        jsonKey = key.concat(".", field);
       }
-    } else if (LocKeys.includes(i)) {
-      const jsonKey = key.concat(i);
-      outputMap[jsonKey] = obj[i];
+      getLocalizeableStrings(objectEntry, jsonKey, outputMap);
 
+    } else if (LocKeys.includes(field)) {
+      jsonKey = key.concat(".", field);
+      const jsonVal = obj[field];
+      if (canLocalize(jsonVal)) {
+        outputMap[jsonKey] = jsonVal;
+      }
       // Check for parameters that should be locked
-      findAndGenerateLockedStringComment(jsonKey, obj[i], outputMap);
+      findAndGenerateLockedStringComment(jsonKey, objectEntry, outputMap);
     }
   }
 }
 
+function canLocalize(text) {
+  const _notAllSpecialCharsRegex = new RegExp(".*[a-zA-Z].*", "g");
```

Comment:
i think you can do this as `new RegExp("[a-z]+", "i")` ?  anything a-z case insensitive anywhere, returning the first match only?  and i think you could create that const outside the function instead of creating a new one over and over?